subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java

```diff
@@ -278,8 +276,7 @@ public void includeBuild(Object rootProject) {
     public void includeBuild(Object rootProject, Action<ConfigurableIncludedBuild> configuration) {
         if (gradle.getParent() == null) {
             File projectDir = getFileResolver().resolve(rootProject);
-            ConfigurableIncludedBuild includedBuild = getIncludedBuildRegistry().addExplicitBuild(projectDir, getNestedBuildFactory());
```

Comment:
@bigdaz can you please verify this change. 

Instead of evaluating the settings of included builds when `includeBuild` is called, we now do it after the script, just before dealing with includes from the command line.

Reply:
LGTM.